Core/Services/CarCategoryService.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using AutoMapper;
+using Core.DTO.CarCategoryDTO;
+using Core.Entities.CarCategoryEntity;
+using Core.Interfaces;
+using Core.Interfaces.CustomService;
+
+namespace Core.Services
+{
+    public class CarCategoryService : ICarCategoryService
+    {
+        private readonly IRepository<CarCategory> _carCategoryRepository;
+        private readonly IMapper _mapper;
+        public CarCategoryService(
+            IRepository<CarCategory> carCategoryRepository,
+            IMapper mapper)
+        {
+            _carCategoryRepository = carCategoryRepository;
+            _mapper = mapper;
+        }
+        
+        public async Task<CarCategoriesListDTO> GetAllCarCategoriesAsync()
+        {
+            var list = await _carCategoryRepository.GetAllAsync();
+            return new CarCategoriesListDTO
+                {CarCategories = _mapper.ProjectTo<CarCategoryDTO>(list.AsQueryable())};
```

Comment:
Why do we need queryable list here?

Reply:
Because _mapper.ProjectTo **source** requires Queryable.

![зображення_2022-05-16_183439893](https://user-images.githubusercontent.com/43933313/168630048-da6732af-687a-4de6-a372-aab8f32f08c4.png)